src/template_parser.rs

```diff
@@ -471,8 +469,17 @@ fn parse_commit_or_change_id_method<'a, I: 'a>(
     self_property: impl TemplateProperty<I, Output = CommitOrChangeId<'a>> + 'a,
     name: Pair<Rule>,
     args_pair: Pair<Rule>,
-    _parse_keyword: &impl Fn(Pair<Rule>) -> TemplateParseResult<PropertyAndLabels<'a, I>>,
+    parse_keyword: &impl Fn(Pair<Rule>) -> TemplateParseResult<PropertyAndLabels<'a, I>>,
 ) -> TemplateParseResult<Property<'a, I>> {
+    let expect_integer_with_default =
```

Comment:
This is about as simple as this can be, right @yuja ? I spent a bit trying to pass an `Option<i64>` to the `TemplatePropertyFn`, but I concluded that it would get pretty messy, and that it'd be cleaner to do what I did here or make separate `TemplatePropertyFn`s for the `Some` and `None` cases.

Reply:
The easiest solution would be to branch if `len_property` is specified.

```toml
let ([], [maybe_len_pair]) = expect_arguments(pair)?;
if Some(len_pair) = maybe_len_pair {
    ...
    chain_properties((self_property, len_property), |(id, len)| ...)
} else {
    chain_properties(self_property, |id| ...)
}
```

We can also add `impl TemplateProperty<C> for Option<P> Output = Option<P::Output>` if that's convenient.

Reply:
Please let me know if this wrapper is useful.
https://github.com/martinvonz/jj/pull/1197

Reply:
That's much nicer than the mess I was afraid of. Yes, I think it's useful, especially for the reason you pointed out in the description. I'm not sure if it's going to be useful immediately, but you never know. Thanks!

Reply:
Of course, I can use it here, which will make it a somewhat documented example.

Reply:
I incorporated the optional arg.